python/export_util.py

```diff
@@ -76,16 +77,82 @@ def convert_to_isoformat(
         return property
 
 
-def get_graph(ctx: mgp.ProcCtx) -> List[Union[Node, Relationship]]:
+def to_duration_iso_format(value: timedelta) -> str:
+    """Converts timedelta to ISO-8601 duration: P<date>T<time>"""
+    date_parts: List[str] = []
+    time_parts: List[str] = []
+
+    if value.days != 0:
+        date_parts.append(f"{abs(value.days)}D")
+
+    if value.seconds != 0 or value.microseconds != 0:
+        abs_seconds = abs(value.seconds)
+        hours = floor(abs_seconds / 3600)
+        minutes = floor((abs_seconds - hours * 3600) / 60)
+        seconds = abs_seconds - hours * 3600 - minutes * 60
+        microseconds = value.microseconds
+
+        if hours > 0:
+            time_parts.append(f"{hours}H")
+        if minutes > 0:
+            time_parts.append(f"{minutes}M")
+        if seconds > 0 or microseconds > 0:
+            microseconds_part = (
+                f".{abs(value.microseconds)}"
+                if value.microseconds != 0
+                else ""
+            )
+            time_parts.append(f"{seconds}{microseconds_part}S")
+
+    date_duration_str = "".join(date_parts)
+    time_duration_str = f'T{"".join(time_parts)}' if time_parts else ""
+
+    return f"P{date_duration_str}{time_duration_str}"
+
+
+def convert_to_isoformat_graphML(
+    property: Union[
+        None,
+        str,
+        bool,
+        int,
+        float,
+        List[Any],
+        Dict[str, Any],
+        timedelta,
+        time,
+        datetime,
+        date,
+    ]
+):
+    if isinstance(property, timedelta):
+        return to_duration_iso_format(property)
+
+    if isinstance(property, (time, date, datetime)):
+        return property.isoformat()
+
+    else:
+        return property
+
+
+def get_graph(
+    ctx: mgp.ProcCtx, graphML: bool
+) -> List[Union[Node, Relationship]]:
     nodes = list()
     relationships = list()
 
     for vertex in ctx.graph.vertices:
         labels = [label.name for label in vertex.labels]
-        properties = {
-            key: convert_to_isoformat(vertex.properties.get(key))
-            for key in vertex.properties.keys()
-        }
+        if graphML:
+            properties = {
+                key: convert_to_isoformat_graphML(vertex.properties.get(key))
+                for key in vertex.properties.keys()
+            }
+        else:
+            properties = {
+                key: convert_to_isoformat(vertex.properties.get(key))
+                for key in vertex.properties.keys()
+            }
```

Comment:
```suggestion
        if graphML:
            properties = {
                key: convert_to_isoformat_graphML(vertex.properties.get(key))
                for key in vertex.properties.keys()
            }
            continue
       
          properties = {
                key: convert_to_isoformat(vertex.properties.get(key))
                for key in vertex.properties.keys()
            }
```

Reply:
but wont continue go to the next vertex in ctx.graph.vertices and lines 157-173 will be skipped?

Reply:
yes but isn't that okay?